core/src/main/java/org/ethereum/beacon/core/BeaconBlockBody.java

```diff
@@ -77,19 +66,38 @@ public BeaconBlockBody(
       ReadList<Integer, Attestation> attestations,
       ReadList<Integer, Deposit> deposits,
       ReadList<Integer, VoluntaryExit> voluntaryExits,
-      ReadList<Integer, Transfer> transfers) {
+      ReadList<Integer, Transfer> transfers,
+      SpecConstants specConstants) {
     this.randaoReveal = randaoReveal;
     this.eth1Data = eth1Data;
     this.graffiti = graffiti;
-    this.proposerSlashings = proposerSlashings;
-    this.attesterSlashings = attesterSlashings;
-    this.attestations = attestations;
-    this.deposits = deposits;
-    this.voluntaryExits = voluntaryExits;
-    this.transfers = transfers;
+    this.proposerSlashings =
+        proposerSlashings.maxSize() == ReadList.VARIABLE_SIZE
+            ? proposerSlashings.cappedCopy(specConstants.getMaxProposerSlashings())
```

Comment:
Capping input list we're avoiding size check of inbound data. It might become a cause for consensus breaks. I think preferable way is to pass data "as-is" onto constructor and verify if capacity matches, otherwise, throw an exception.

Reply:
it's not a cap, it's a checking whether its already limited list or not yet

Reply:
got it!